vnext/Microsoft.ReactNative/ReactApplication.cpp

```diff
@@ -179,34 +183,13 @@ void ReactApplication::OnCreate(IActivatedEventArgs const &e) {
       // Restore the saved session state only when appropriate, scheduling the
       // final launch steps after the restore is complete
     }
-
-    if (!isPrelaunchActivated) {
-      if (rootFrame.Content() == nullptr) {
-        // When the navigation stack isn't restored navigate to the first page,
-        // configuring the new page by passing required information as a
-        // navigation parameter
-        content = m_delegate.OnCreate(args);
-        rootFrame.Content(content);
-      }
-
-      // Place the frame in the current Window
-      Window::Current().Content(rootFrame);
-      // Ensure the current window is active
-      Window::Current().Activate();
-    }
-  } else {
-    if (!isPrelaunchActivated) {
-      if (rootFrame.Content() == nullptr) {
-        // When the navigation stack isn't restored navigate to the first page,
-        // configuring the new page by passing required information as a
-        // navigation parameter
-        content = m_delegate.OnCreate(args);
-        rootFrame.Content(content);
-      }
-      // Ensure the current window is active
-      Window::Current().Activate();
-    }
   }
+  //
+  //
+  ApplyArguments(Host(), args.c_str());
+
+  // Nudge the ReactNativeHost to create the instance and wrapping context
+  Host().ReloadInstance();
```

Comment:
>Host().ReloadInstance(); [](start = 2, length = 24)

This seems to be the right place. Though, we can it event earlier if we could. Ideally it should be done as soon as we have all settings ready.